src/json.hpp.re2c

```diff
@@ -65,6 +65,12 @@ Class @ref nlohmann::basic_json is a good entry point for the documentation.
     #endif
 #endif
 
+// disable float-equal warnings on GCC/clang
+#if defined(__clang__) || defined(__GNUC__) || defined(__GNUG__)
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wfloat-equal"
+#endif
+
```

Comment:
This is necessary to suppress floating point warnings under gcc/clang.  The only real alternatives are to:

1. Accept the warnings - they are only warnings after all (ugly though)
2. Change the build settings in Travis.yml so that -Wfloat-equal is omitted at the command line - this works for us but if the user decides to compile with the warning, and doesn't understand floating point, then they may come to the conclusion that this is a bug in our code
3. Add the #pragma without the conditional block - this creates its own warning about unknown #pragma under VS2015
4. Reinstate approx()
5. Come up with some other means of avoiding the warning if such a thing exists

I will revert this for now and add it to its own pull request as it is really a separate issue.